src/uwtools/resources/jsonschema/esg-grid.jsonschema

```diff
@@ -0,0 +1,84 @@
+{
+  "properties": {
+    "esg_grid": {
+      "additionalProperties": false,
+      "properties": {
+        "execution": {
+          "$ref": "urn:uwtools:execution"
+        },
+        "namelist": {
+          "additionalProperties": false,
+          "anyOf": [
+            {
+              "required": [
+                "base_file"
+              ]
+            },
+            {
+              "required": [
+                "update_values"
+              ]
+            }
+          ],
+          "properties": {
+            "base_file": {
+              "type": "string"
+            },
+            "update_values": {
+              "properties": {
+                "regional_grid_nml": {
+                  "additionalProperties": {
+                    "type": [
+                      "array",
+                      "boolean",
+                      "number",
+                      "string"
+                    ]
+                  },
+                  "properties": {
+                    "delx": {
+                      "type": "number"
+                    },
+                    "dely": {
+                      "type": "number"
+                    },
+                    "lx": {
+                      "type": "number"
+                    },
+                    "ly": {
+                      "type": "number"
+                    },
+                    "pazi": {
+                      "type": "number"
+                    },
+                    "plat": {
+                      "type": "number"
+                    },
+                    "plon": {
```

Comment:
I don't know which is correct, but this key is listed as `plot` in the sample config under `docs/shared`. If this is correct, and if the sample config was tested with `uw esg_grid validate` and passed, then the permissive `additionalProperties` above would be why. But I guess this is probably `plot` given the ordering...

We might also consider making these values required if they're required to be set in the namelist.

Reply:
Based on the example code, they should all be required. I'll add a section for this and `plon` is the correct key name based on the [documentation](https://github.com/ufs-community/UFS_UTILS/blob/ef3a1d2c39d0620692ec1bb8026d587cb40dca4f/sorc/grid_tools.fd/regional_esg_grid.fd/pesg.f90#L1820).

Reply:
Wondering a bit more on 

> We might also consider making these values required if they're required to be set in the namelist.

When we want them to be required, we'd like them to be required in the `namelist` object, but they don't necessarily need to be supplied by the user every time they want to update their base files. So in the `namelist` object we should be adding a `required` object and in addition to having the `namelist` options in the `update_values` object as optional, we should also be adding them in the `namelist` object as required.

Did I understand your suggestion right? Or do we simply make these required in the `update_values` object and update the schema tests accordingly. One test might look like: